### DISCUSSION TITLE
Bagaimana Cara Sorting Hasil sitemap.xml?

### DISCUSSION THREAD

**Discussion by idkarirku** [mecha-cms/mecha#254] [Q&A]
hampir 1 tahun ane coba memahami mecha, baru sadar sitemap mecha tidak disorting by time. wkwkwk.

Tolong Bantu kang, Bagaimana cara sorting by time hasil sitemap.xml?

**Comment by taufik-nurrohman** [comment]
Sebenarnya posisi URL [nggak ngaruh](https://www.sitemaps.org/faq.html#faq_url_position) bagi mesin pencari mas. Karena berkas ini tujuannya cuma untuk register daftar URL saja.

> **Q: Does position of a URL in a Sitemap influence its use?**
> No. The position of a URL in the Sitemap is not likely to impact how it is used or regarded by search engines.

Tapi kalau mas memang harus melakukan itu ada si caranya. Pertama-tama setiap item harus diparse menjadi objek `Page` dulu untuk mendapatkan nilai properti `time`, setelah itu baru deh bisa disortir berdasarkan waktu.

~~~ .php
foreach (\Pages::from($folder, 'page', true)->sort([-1, 'time']) as $k => $v) {
    $k = $v->path; // Ambil nilai `path` dari objek yang telah disortir
    // Selebihnya sama…
    if (!$kk = \exist([
        $k . '.archive',
        $k . '.page'
    ], 1)) {
        continue;
    }
    $content .= '<url>';
    $content .= '<loc>' . \Hook::fire('link', ['/' . ($r = \strtr(\strtr($k, [\LOT . \D . 'page' . \D => ""]), \D, '/'))]) . '</loc>';
    $priority = \b(1 - (\substr_count($r, '/') * .1), [.5, 1]); // `0.5` to `1.0`
    $content .= '<lastmod>' . \date('c', \filemtime($kk)) . '</lastmod>';
    $content .= '<changefreq>monthly</changefreq>';
    $content .= '<priority>' . $priority . '</priority>';
    $content .= '</url>';
}
~~~

Yang paling utama itu ada di bagian ini saja:

~~~ .php
foreach (\Pages::from($folder, 'page', true)->sort([-1, 'time']) as $k => $v) {
    $k = $v->path; // Ambil nilai `path` dari objek yang telah disortir
    // …
~~~

Edit di bagian ini:

https://github.com/mecha-cms/x.sitemap/blob/557da00699990a958ee38aa587e9d95b2f01c82e/index.php#L42-L57

**Comment by taufik-nurrohman** [comment]
Oiya lupa untuk bagian sitemap per folder:

~~~ .php
foreach (\Pages::from($folder, 'page', true)->sort([-1, 'time']) as $k => $v) {
    $k = $v->path; // Ambil nilai `path` dari objek yang telah disortir
    $content .= '<sitemap>';
    $content .= '<loc>' . \Hook::fire('link', ['/' . \strtr(\strtr($k, [\LOT . \D . 'page' . \D => ""]), \D, '/') . '/sitemap.xml']) . '</loc>';
    $content .= '<lastmod>' . \date('c', \filemtime($kk)) . '</lastmod>';
    $content .= '</sitemap>';
}
~~~

Edit di bagian ini:

https://github.com/mecha-cms/x.sitemap/blob/557da00699990a958ee38aa587e9d95b2f01c82e/index.php#L64-L79
